src/commands/extract-inline-scripts.yml

```diff
@@ -0,0 +1,93 @@
+description: >
+  Extract shell script snippets from `run` blocks within the packed
+  source  code (single file) of an orb and store each snippet as a
+  temporary file to be checked by the shellcheck orb. Requires Bash
+  v4.0 or later.
+
+parameters:
+  target-filepath:
+    type: string
+    default: orb.yml
+    description: >
+      Where is the YAML file containing orb source code? Defaults to a
+      file named `orb.yml` in the working directory
+
+  output-dir:
+    type: string
+    default: scripts
+    description: >
+      Where should scripts be outputted? Defaults to a directory named
+      `scripts` (it will be created if it does not exist) in the working
+      directory
+
+  run-pack:
+    type: boolean
+    default: true
+    description: >
+      Pack a destructured orb source directory for this orb command to use?
+
+  pack-source:
+    type: string
+    default: src
+    description: Path to the root folder of an unpacked orb
+
+  pack-destination:
+    type: string
+    default: orb.yml
+    description: Path including the filename to output the packed orb
+
+  debug:
+    type: boolean
+    default: false
+    description: >
+      Extra output for orb developers
+
+steps:
+  - build-tools/install-ci-tools
```

Comment:
It looks to me like the plan here is to use the shellcheck orb as a linter for other orbs as well as for `config.yml` files?

I think the way you have implemented this is inside out - with this way this is written, the `shellcheck` orb has full knowledge of orbs, the circleci command line tools, etc. The shellcheck orb should just be responsible for running shellcheck - it should be the responsibilty of another orb to pass the required scripts to the `shellcheck` orb for checking IMO.

Reply:
just orbs, i was thinking @marcomorain

Reply:
hmmmmm i see what you're saying, but im not sure where that functionality would live then, or what that orb would be... i suppose orb-tools or build-tools could have a command to extract shell scripts from yaml, but i'm not sure what else that command would get used for